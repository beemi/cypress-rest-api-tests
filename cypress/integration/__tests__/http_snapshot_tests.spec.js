describe("POST method tests", function () {

    const chance = require("chance");

    context("POST create resources", function () {

        it('should be able to create resources,200 OK', function () {

            let user_id = chance().integer({min: 1, max: 20000});
            let id;

            cy.api({
                method: "POST",
                url: "/posts",
                headers: {
                    "Content-type": "application/json; charset=UTF-8"
                },
                body: JSON.stringify({
                    title: chance().word(),
                    body: chance().paragraph(),
                    userId: user_id,
                })
            }).its("body").toMatchSnapshot({
                "ignoreExtraFields": false,
                "ignoreExtraArrayItems": false,
                "normalizeJson": true,
            });
        });
    })

});
