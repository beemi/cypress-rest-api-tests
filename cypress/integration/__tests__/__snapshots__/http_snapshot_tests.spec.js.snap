exports[`POST method tests > POST create resources > should be able to create resources,200 OK #0`] =
{
  "body": "Zihifew ziknupsat vibnu fuimaw jaic lazilu upu vadzoz vadip jejlat asaido jatam sis homtepiv po weahza mifukaabu hunkiral. Borliwlut boz be tiunizab pinzaja edi uddew powta jogiz getij udnaz zaine icekivku aware kuga ibepzu. Vekog zogwiwpeh gipporit zutbovci cisaslu illa hasfi mobadiwo ce maethir godli unla mircak uljevpu. Uho geola cet mihiefa zuenu vog watucnig rufruj av bidwi hu rugi wehjoru jidki gesge ju hubuotu.",
  "id": 101,
  "title": "dal",
  "userId": 8009
};
